### PR TITLE
Set Default Source in chocolatey

### PR DESCRIPTION
**Pull request by DamianBis**
This is for #24.

The test is a bit of a hacky way to find the sources. but going through the XML config was preferable to doing a select-string/regex on the output to chocolatey sources list.

It compares on name not URL. so if name already exists but the URL is different it won't overwrite it will return true. this can be improved later if required.

### CODE REVIEW

File: DSCResources/cChocoSource/cChocoSource.psm1 [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	if($Ensure -eq "Present")
+	{
+		if($priority -eq $null)
+		{
+			choco sources add -n"$name" -s"$source"
+		}
+		else
+		{
+			choco sources add -n"$name" -s"$source" --priority=$priority
+		}
```
Comment (ferventcoder) [inline DSCResources/cChocoSource/cChocoSource.psm1:88]: --user --password ? 

Otherwise looks good.

File: DSCResources/cChocoSource/cChocoSource.psm1 [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    )
+
+    Write-Verbose "Start Test-TargetResource"
+	$exe = (get-command choco).Source
+	$chocofolder = $exe.Substring(0,$exe.LastIndexOf("\"))
+
+	if( $chocofolder.EndsWith("bin") )
+	{
+		$chocofolder = $chocofolder.Substring(0,$chocofolder.LastIndexOf("\"))
+	}
```
Comment (ferventcoder) [inline DSCResources/cChocoSource/cChocoSource.psm1:135]: Why not start with `$env:ChocolateyInstall`?

File: DSCResources/cChocoSource/cChocoSource.psm1 [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	else
+	{
+		return $true
+	}
+}
+
+
+##region - chocolately installer work arounds. Main issue is use of write-host
+##attempting to work around the issues with Chocolatey calling Write-host in its scripts. 
+function global:Write-Host
```
Comment (ferventcoder) [inline DSCResources/cChocoSource/cChocoSource.psm1]: Why is this necessary here? Choco source is run completely from the executable and never calls powershell. If this has some other purpose, perhaps we should move it to a different file?

Comment (DamianBis) [inline DSCResources/cChocoSource/cChocoSource.psm1]: You're right. i copied from one of the others and forgot to remove the extra function

File: DSCResources/cChocoSource/cChocoSource.psm1 (hunk 2 above)
Comment (DamianBis) [inline DSCResources/cChocoSource/cChocoSource.psm1:135]: Completely forgot that $env:ChocolateyInstall gets set. I'll put another commit in that uses that and if it's null then looks for it like this.

File: DSCResources/cChocoSource/cChocoSource.psm1 (hunk 1 above)
Comment (ebekker) [inline DSCResources/cChocoSource/cChocoSource.psm1:88]: @DamianBis, I think @ferventcoder's suggestion for optional credentials would be useful for authenticated feeds.  You can allow either a pair of `string` and `securestring` parameters for username and password, or a single `pscredential`.

Comment (DamianBis) [inline DSCResources/cChocoSource/cChocoSource.psm1:88]: Only way i can think of is using plain text password. as i need to pass it through to chocolatey as plaintext.

Comment (dlwyatt) [inline DSCResources/cChocoSource/cChocoSource.psm1:88]: Passwords in DSC resources should always be PSCredential objects.  That's the only thing that will trigger encryption of the password in the MOF file (if you've set up certificates for the node).

Inside the resource, you can fetch the plain-text version of the password easily with `$Credential.GetNetworkCredential().Password`

Comment (DamianBis) [inline DSCResources/cChocoSource/cChocoSource.psm1:88]: Ahh didn't know how to get plaintext with $Credential.GetNetworkCredential().Password. I'll write up some code and commit today.

**Comment by ebekker** [comment]
Changes look good, thanks!  And thanks everyone for reviewing and weighing in to refine the PR!
